### ISSUE TITLE
What is the difference between it and SDFT?

### ISSUE THREAD

**Issue by nonu116** [lasgroup/SDPO#6]
SDFT : https://arxiv.org/abs/2601.19897

* Both use on-policy rollout from the student.
* Both combine ICL (Independent Classification) to construct the teacher model from the student model (the current training model).
* Both's loss function minimizes the KL divergence between the student and teacher models.

What is their essential difference? Is it only in the construction method of ICL? SDFT uses demonstrations from the dataset to construct ICL, while SDPO uses error messages (such as runtime errors from a code environment) from the environment or other correctly sampled data from the group.

**Comment by Gengsheng-Li** [comment]
same questions. them seem same except of the demonstartion

**Comment by kirawi** [comment]
https://github.com/huggingface/trl/pull/4941#issuecomment-3831889829

> […] it is 1to1 the same algorithm. The only difference is where the teacher prompts come from. In SDPO, the teacher prompts are created "online" using generated trajectories that are marked as correct by the environment and any other rich signal returned by the environment.

**Comment by jonhue** [comment]
Hi all, thanks for the question. As mentioned in https://github.com/huggingface/trl/pull/4941#issuecomment-3831889829, the losses are identical. The only difference is where the data for the self-teacher comes from.
